Fwa 004 handle data from json

heupotato opened the pull request.

heupotato committed:
Add file mock json data

heupotato committed:
Create models

heupotato committed:
Generate models by json_model package

heupotato committed:
Update pubspec.yaml

heupotato committed:
Update assets.dart

heupotato committed:
Update home_screen.dart

heupotato committed:
Update date_formatter.dart

heupotato committed:
Parse data to days and week, extract data according to day and week

heupotato committed:
Update home_screen.dart

heupotato committed:
Update weather_data_repository.dart

lngyeen commented on Flutter_Weather/assets/jsons/mock_json_data.json:
```diff
@@ -0,0 +1,902 @@
+﻿{
```
File name should describe file contents. (Should be 'mock_weather.json')

lngyeen commented on Flutter_Weather/jsons/mockjsondata.json:
```diff
@@ -0,0 +1,5 @@
+{
```
This is a json data model file,  not a mock data file. So the name should be 'weather.json'

lngyeen commented on Flutter_Weather/lib/models/weather_data.dart:
```diff
@@ -0,0 +1,145 @@
+}
+
+@immutable
+class Wind10m {
```
Should create json file 'wind10m.json' put it in jsons/ folder. So json_to_model will generate separated files for each model. Easy to maintain code.

lngyeen commented on Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:
```diff
@@ -0,0 +1,58 @@
+extension ProcessMockJsonData on Mockjsondata{
+  //int getInitTime() => int.parse(init.substring(init.length - 2, init.length));
+
+  DateTime getInitDate() => DateFormatter.yearMonthDayHour(init);//DateFormatter.yearMonthDay(init.substring(0, init.length -2));
```
Should be get property, not function: 'DateTime get initDate => /...'

lngyeen commented on Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:
```diff
@@ -0,0 +1,58 @@
+
+  DateTime getInitDate() => DateFormatter.yearMonthDayHour(init);//DateFormatter.yearMonthDay(init.substring(0, init.length -2));
+
+  DateTime getTrueDate(int timePoint) => this.getInitDate().add(Duration(hours: timePoint));
```
Same above, should be get property

lngyeen commented on Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:
```diff
@@ -0,0 +1,58 @@
+
+  DateTime getTrueDate(int timePoint) => this.getInitDate().add(Duration(hours: timePoint));
+
+  List<DateTime> getTrueDates(List<WeatherData> dataList)
```
same above

lngyeen commented on Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:
```diff
@@ -0,0 +1,58 @@
+  List<DateTime> getTrueDates(List<WeatherData> dataList)
+   => dataList.map((data) => this.getTrueDate(data.timepoint)).toList();
+
+  List<String> getWeekDays(List<WeatherData> dataList)
```
same above

lngyeen commented on Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:
```diff
@@ -0,0 +1,58 @@
+    return weekData;
+  }
+
+  List<WeatherData> getDayData(){
```
Day data some time has only 1 or 2 time points, not always 8. 
For example: Weather data was created at 6PM UTC time, so today, we only have 2 weather time points [3, 6]. (Real time of each weather item will be:  WeatherData's initTime + item's time point).
 You should create a class 'DayWeather' to handle this.

lngyeen commented on Flutter_Weather/lib/storage/json_repositories/weather_data_repository.dart:
```diff
@@ -0,0 +1,58 @@
+    return indexes;
+  }
+
+  List<WeatherData> getWeekData(){
```
Should create class 'WeekWeather' contains List of DayWeather to handle data for a week.

lngyeen commented on Flutter_Weather/lib/services/date_formatter.dart:
```diff
@@ -18,6 +18,15 @@ class DateFormatter {
 
-  static DateTime yearMonthDayHour(String initDate) =>
-      DateFormat('yyyy MM dd hh').parse(initDate, true);
+  static DateTime yearMonthDayHour(String initDate){
```
Should move to Weather class extension, This special value handling occurs only in that class, it is specific to that class.

heupotato committed:
Update json models

heupotato committed:
Update date_formatter.dart

heupotato committed:
Update weather_data_repository.dart

Resolve:
- Add getter if possible
- Modify getDayData method
- Moe yearMonthDayHour method

heupotato committed:
Update home_screen.dart

heupotato committed:
Update weather_data_repository.dart

Add more timepoints if today's timepoints are less than 5

lngyeen commented:
Resolve conflicts.

heupotato commented:
Conflicts resolved
